public/app/features/dashboard/components/DashboardSettings/JsonEditorSettings.tsx

```diff
@@ -30,25 +34,33 @@ export const JsonEditorSettings: React.FC<Props> = ({ dashboard }) => {
         settings, layout, queries, and so on.
       </div>
 
-      <div>
-        <AutoSizer disableHeight>
-          {({ width }) => (
+      <div className={styles.editWrapper}>
+        <AutoSizer>
+          {({ width, height }) => (
             <CodeEditor
               value={dashboardJson}
               language="json"
               width={width}
-              height="500px"
-              showMiniMap={false}
+              height={height}
+              showMiniMap={true}
+              showLineNumbers={true}
               onBlur={onBlur}
             />
           )}
         </AutoSizer>
       </div>
       {dashboard.meta.canSave && (
-        <Button className="m-t-3" onClick={onClick}>
-          Save changes
-        </Button>
+        <HorizontalGroup>
+          <Button onClick={onClick}>Save changes</Button>
+        </HorizontalGroup>
       )}
     </>
   );
 };
+
+const getStyles = stylesFactory((theme: GrafanaTheme) => ({
```

Comment:
The stylesFactory is deprecated so you should remove it and use the `useStyles|useStyles2` hooks instead. Also, we're moving to GrafanaThemeV2 so it would be great if you can use that instead too.
```suggestion
const getStyles = (theme: GrafanaThemeV2) => ({
```